Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -324,24 +327,30 @@ internal void SetHttpClient(HttpClient client)
 
         protected HttpClient CreateOrUseCachedHttpClient()
         {
-            if (httpClient == null)
+            // The httpClient is the property that will be used for tests only.
+            if (httpClient != null)
+            {
```

Comment:
shouldn't we be using this. as a prefix here? to make it clear we're referencing instance data on the class?

i like this pattern, *or* we could use m_ as a prefix to make that distinction clear. what do we do elsewhere?